backend/Services/DataSanitizer.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use Exception;
+
+define("SANITIZE_NONE", (1 << 0));
+define("SANITIZE_WHITESPACE_TRIM", (1 << 1));
+define("SANITIZE_WHITESPACE_REMOVE", (1 << 2));
+define("SANITIZE_WHITESPACE_TO_DASH", (1 << 3));
+define("SANITIZE_TO_LOWER", (1 << 4));
+define("SANITIZE_TO_UPPER", (1 << 5));
+define("SANITIZE_TO_CAPITALIZE_FIRST", (1 << 6));
+define("SANITIZE_CAPITALIZE_WORDS", (1 << 7));
+define("SANITIZE_NO_ACCENTS", (1 << 8));
+
+abstract class DataSanitizer
+{
+    public function sanitize(
+        string $value,
+        int $flags = SANITIZE_WHITESPACE_TRIM,
+        string $arraySeparator = null,
+        int $decimals = null
+    ): mixed {
+        if ($flags & SANITIZE_NO_ACCENTS) {
+            $value = normalizer_normalize($value);
+        }
+
+        if ($flags & SANITIZE_WHITESPACE_REMOVE) {
+            $value = preg_replace("/\s+/", "", $value);
+        } else {
+            if ($flags & SANITIZE_WHITESPACE_TO_DASH) {
+                $value = preg_replace("/\s+/", "-", $value);
+            } else {
+                if ($flags & SANITIZE_WHITESPACE_TRIM) {
+                    $value = trim(preg_replace("/\s+/", " ", $value));
+                }
+            }
+        }
+
+        if ($flags & SANITIZE_TO_LOWER) {
+            $value = strtolower($value);
+        } else {
+            if ($flags & SANITIZE_TO_UPPER) {
+                $value = strtoupper($value);
+            } else {
+                if ($flags & SANITIZE_TO_CAPITALIZE_FIRST) {
+                    $value = ucfirst($value);
+                } else {
+                    if ($flags & SANITIZE_CAPITALIZE_WORDS) {
+                        $value = ucwords($value);
+                    }
+                }
+            }
+        }
+
+        if ($arraySeparator !== null) {
+            $trimmedArray = [];
+            foreach (explode($arraySeparator, $value) as $item) {
+                if ($decimals !== null) {
+                    $item = preg_replace("/\s+/", "", $item);
+                    array_push($trimmedArray, floatval(number_format(floatval($item), $decimals)));
+                } else {
+                    $item = trim($item);
+                    array_push($trimmedArray, $item);
+                }
+            }
+            return $trimmedArray;
+        }
+        return $value;
+    }
+
+    public function validateField(
```

Comment:
Maybe we could narrow the responsibility of `DataSanitizer` by splitting this class into two? 
`DataSanitizer` and `DataValidator`?

The method signature suggests that we're validating a bunch of stuff here (dataName, id, requirement, some flags, decimals, etc.) but in fact, we're only checking the requirement of a given field, the rest of the arguments are for sanitizing data.